### PR TITLE
Removed AutoSubscribePlainMessages as settings option

### PR DESCRIPTION
**Pull request by dvdstelt**
Related to this issue : https://github.com/Particular/NServiceBus/issues/2318

Doco PR: https://github.com/Particular/docs.particular.net/pull/997

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@Particular/nservicebus please review.

**Comment by danielmarbach** [comment]
Looks good from a good perspective. I made a comment on the doco PR.

File: src/NServiceBus.Core/obsoletes.cs [hunk 1]
```diff
@@ -832,6 +832,15 @@ public void DoNotRequireExplicitRouting()
         {
             throw new NotImplementedException();
         }
+
+        [ObsoleteEx(
+            TreatAsErrorFromVersion = "6",
+            RemoveInVersion = "7",
+            Message = "Encourages bad practices. IBusSession.Subscribe should be explicitely used.")]
```
Comment (andreasohlund) [inline src/NServiceBus.Core/obsoletes.cs]: Typo

**Comment by mikeminutillo** [comment]
This issue does not currently appear on the waffle board during syncs. Please find (or create) and issue on the [waffle board](https://waffle.io/particular/progress-strategic) for this and use the waffle ["Connect" keyword](https://github.com/waffleio/waffle.io/wiki/FAQs#prs-connect-keywords) to associate this issue with it.

File: src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt [hunk 2]
```diff
@@ -1070,6 +1070,8 @@ namespace NServiceBus.AutomaticSubscriptions.Config
 {
     public class AutoSubscribeSettings
     {
+        [System.ObsoleteAttribute("Encourages bad practices. IBusSession.Subscribe should be explicitly used. Will b" +
```
Comment (indualagarsamy) [inline src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt]: Encourages bad practices. ? This message isn't very clear. Can you add more context?

**Pull request merged**